### PR TITLE
GH-34410: [Python] Allow chunk sizes larger than the default to be used

### LINKED ISSUE #34410: [Python][C++] No longer possible to specify higher chunksize than the default for Parquet writing
### Describe the bug, including details regarding any error messages, version, and platform.

See https://github.com/apache/arrow/issues/34374#issuecomment-1449926603 for context

https://github.com/apache/arrow/pull/34281 changed the default row group size (`chunksize` in the C++ WriteTable API). However, that PR changed `DEFAULT_MAX_ROW_GROUP_LENGTH`, which doesn't set the _default_ `chunksize`, but actually caps the max chunk (row group) size regardless of the user-specified `chunksize`.

It seems this constant is used both for setting this max upper cap, and as the default values for `chunk_size` in `WriteTable`. I assume we will have to distinguish those two meanings.

### Component(s)

C++, Parquet, Python

### PR DESCRIPTION
**Pull request by westonpace**
### Rationale for this change

We changed the default chunk size from 64Mi rows to 1Mi rows.  However, it turns out that this property was being treated not just as the default but also as the absolute max.  So it was no longer possible to specify chunk sizes larger than 1Mi rows.  This change separates those two things and restores the max to 64Mi rows.

### What changes are included in this PR?

Pyarrow will now set the `ParquetWriterProperties::max_row_group_length` to 64Mi when constructing a parquet writer.

### Are these changes tested?

Yes.  Unit tests are added.

### Are there any user-facing changes?

No.  The previous change #34281 changed two defaults (absolute max and default).  This PR restores the absolute max back to what it was before.  So it is removing a user-facing change.
* Closes: #34410

### CODE REVIEW

File: python/pyarrow/tests/parquet/test_parquet_writer.py [hunk 1]
```diff
@@ -202,6 +203,39 @@ def test_parquet_writer_write_wrappers(tempdir, filesystem):
     tm.assert_frame_equal(result, df)
 
 
+@pytest.mark.pandas
```
Comment (wjones127) [inline python/pyarrow/tests/parquet/test_parquet_writer.py:206]: Perhaps also:

```suggestion
@pytest.mark.large_memory
@pytest.mark.pandas
```

**Review by wjones127 — CHANGES_REQUESTED** [review]
Overall looks good. One suggestion.

File: python/pyarrow/tests/parquet/test_parquet_writer.py (hunk 1 above)
Comment (westonpace) [reply python/pyarrow/tests/parquet/test_parquet_writer.py:207]: Good idea

**Review by wjones127 — APPROVED** [review]

File: python/pyarrow/_parquet.pyx [hunk 2]
```diff
@@ -1597,6 +1597,15 @@ cdef shared_ptr[WriterProperties] _create_writer_properties(
...
             (<FileEncryptionProperties>encryption_properties).unwrap())
 
+    # For backwards compatibility reasons we cap the maximum row group size
+    # at 64Mi rows.  This could be changed in the future, though it would be
+    # a breaking change.
+    #
+    # The user can always specify a smaller row group size (and the default
+    # is smaller) when calling write_table.  If the call to write_table uses
+    # a size larger than this then it will be latched to this value.
+    props.max_row_group_length(64*1024*1024)
```
Comment (mapleFU) [inline python/pyarrow/_parquet.pyx:1607]: does this causing the bug in https://github.com/apache/arrow/issues/35859#issuecomment-1583614922 ?

**Pull request merged**